Automattic/jetpack
# My Jetpack: Expose standalone plugin information on My Jetpack products initial state

Fixes #29902.

## Proposed changes:

* Exposes a `standalone_plugin_info` property on every product of the My Jetpack initial state data, containing three self-explanatory flags related to standalone plugin status:
   * `has_standalone_plugin`
   * `is_standalone_installed`
   * `is_standalone_active`
* Introduces a flag into the `Product` class to mark products that has a standalone plugin
* Marks `VideoPress`, `Social`, `Search` and `Backup` as products that have a standalone plugin

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Launch a fresh website containing only Jetpack plugin
* Go to `Jetpack > My Jetpack` and open the browser console
* Inspect the content for the `myJetpackInitialState.products.items.videopress` variable
* It should contains the `standalone_plugin_info` property:

<img width="347" alt="Screen Shot 2023-04-06 at 16 25 49" src="https://user-images.githubusercontent.com/6760046/230476216-693e7176-24fe-466f-acf3-d14aefb1c85e.png">

* Inspect the value for the `standalone_plugin_info` property for VideoPress (`myJetpackInitialState.products.items.videopress.standalone_plugin_info`)
* It should show that VideoPress has a standalone plugin, that it's not installed and that it's not active:

<img width="658" alt="Screen Shot 2023-04-06 at 16 58 12" src="https://user-images.githubusercontent.com/6760046/230481808-e7496589-3681-4d63-9c57-7274a8b876c3.png">

* Now install Jetpack VideoPress plugin but does not activate it (install a released version using `Plugins > Add new` instead of Jetpack Beta so you can control if the plugin is active or not)
* Go back to `My Jetpack` screen and check the `myJetpackInitialState.products.items.videopress.standalone_plugin_info` property again
* It should show that VideoPress is now installed but not active:

<img width="651" alt="Screen Shot 2023-04-06 at 16 43 11" src="https://user-images.githubusercontent.com/6760046/230479029-2676360b-bf99-4c90-ae5a-84776faf2fba.png">

* Now activate the Jetpack VideoPress plugin
* Go back to `My Jetpack` screen and check the `myJetpackInitialState.products.items.videopress.standalone_plugin_info` property again
* It should show that VideoPress is now active:

<img width="656" alt="Screen Shot 2023-04-06 at 16 56 00" src="https://user-images.githubusercontent.com/6760046/230481300-886cfa8f-5d5f-4756-ad3b-ff770bdab254.png">

* You can test for `VideoPress`, `Social`, `Search` or `Backup`
